megamek/src/megamek/codeUtilities/MathUtility.java

```diff
@@ -21,57 +21,117 @@
+     * a f of 0d will return the minimum, a f of 1d will return the maximum. Otherwise, this will
+     * return the rounded integer between the two points
+     * @note we assume this is used correctly, and thus do not check if the min and max are handled
+     * incorrectly
      */
     public static int lerp(final int min, final int max, final double f) {
         // The order of operations is important here, to not lose precision
         return (int) Math.round(min * (1d - f) + max * f);
     }
 
     /**
-     * @return double linear interpolation value between min and max
+     * @param min the minimum value
+     * @param max the maximum value
+     * @param f location factor between the two points
+     * @return double graphical linear interpolation value between min and max
+     * a f of 0d will return the minimum, a f of 1d will return the maximum. Otherwise, this will
+     * return the double value between the two points
+     * @note we assume this is used correctly, and thus do not check if the min and max are handled
+     * incorrectly
      */
     public static double lerp(final double min, final double max, final double f) {
-        // The order of operations is important here, to not lose precision
         return min * (1d - f) + max * f;
     }
 
     /**
-     * @return float linear interpolation value between min and max
+     * @param min the minimum value
+     * @param max the maximum value
+     * @param f location factor between the two points
+     * @return float graphical linear interpolation value between min and max
+     * a f of 0f will return the minimum, a f of 1f will return the maximum. Otherwise, this will
+     * return the float value between the two points
+     * @note we assume this is used correctly, and thus do not check if the min and max are handled
+     * incorrectly
      */
     public static float lerp(final float min, final float max, final float f) {
-        // The order of operations is important here, to not lose precision
         return min * (1f - f) + max * f;
     }
 
     /**
-     * @return long linear interpolation value between min and max
+     * @param min the minimum value
+     * @param max the maximum value
+     * @param f location factor between the two points
+     * @return long rounded graphical linear interpolation value between min and max
+     * a f of 0d will return the minimum, a f of 1d will return the maximum. Otherwise, this will
+     * return the rounded long between the two points
+     * @note we assume this is used correctly, and thus do not check if the min and max are handled
+     * incorrectly
      */
-    public static long lerp(final long min, final long max, final long f) {
+    public static long lerp(final long min, final long max, final double f) {
         // The order of operations is important here, to not lose precision
-        return min * (1L - f) + max * f;
+        return Math.round(min * (1L - f) + max * f);
```

Comment:
Every time I see a change to a "deep library" function like this, I get scared. Do we have any idea where it's used? And if so, can we get a short write-up of stuff that can be tested from the user side to see if the change "affects" anything?

Reply:
According to my IDE the lerps are not currently used at all. I guess it's safe to merge.